bkbit/scripts/gfftranslator.py

```diff
@@ -1,203 +1,682 @@
-
-            if row['label'] not in genomeAnnotObjects:
-                genomeAnnotObjects[row['label']] = kbmodel.GenomeAnnotation(id = 'bican:assembly-' + row['label'])
-
-            if row['taxon_local_unique_identifier'] not in orgTaxonObjects:
-                orgTaxonObjects[row['taxon_local_unique_identifier']] = kbmodel.OrganismTaxon(id = 'NCBITaxon:' + str(row['taxon_local_unique_identifier']))
-
-            fname = 'gene_annotation_%s-%s-%s.csv' % (row['authority'], str(row['taxon_local_unique_identifier']), str(row['version']))
-            file = os.path.join(output_dir, fname)
-            if not os.path.isfile(file):
-                parse_data(df, row['authority'], row['label'], row['taxon_local_unique_identifier'], row['version'], row['gene_identifier_prefix'], output_dir)
-            print(f'Parsed DF containing GeneAnnotation class attribute values is saved here: {file}')
-            
-            annotations = create_gene_annotation_objects(file, genomeAnnotObjects[row['label']], orgTaxonObjects[row['taxon_local_unique_identifier']])
-            output_ser = os.path.join(output_dir, gene_name +'.json')
-            serialize_annotation_collection(annotations.annotations, output_ser)
-            print(f'Serialized GeneAnnotation objects saved here: {output_ser}')
-            
-    orgTaxon_genomeAnnot_file = os.path.join(output_dir, 'orgTaxon_genomeAnnot.json')
-    annotations = []
-    annotations.extend(list(orgTaxonObjects.values()))
-    annotations.extend(list(genomeAnnotObjects.values()))
-    serialize_annotation_collection(annotations, orgTaxon_genomeAnnot_file)
-    print(f'Serialized OrganismTaxon and GenomeAnnotation objects saved here: {orgTaxon_genomeAnnot_file}')
-
-if __name__ == '__main__':
+                value = attributes[attribute_name].pop()
+                if value.find(",") != -1:
+                    logger.warning(
+                        'Line %s: %s not set for this row\'s GeneAnnotation object due to value of %s attribute containing ",".',
+                        curr_line_num,
+                        attribute_name,
+                        attribute_name,
+                    )
+                    value = None
+        else:
+            logger.warning(
+                "Line %s: %s not set for this row's GeneAnnotation object due to missing %s attribute.",
+                curr_line_num,
+                attribute_name,
+                attribute_name,
+            )
+        return value
+
+    def __resolve_ncbi_gene_annotation(self, new_gene_annotation, curr_line_num):
+        """
+        Resolves conflicts between existing and new gene annotations based on certain conditions.
+
+        Args:
+            new_gene_annotation (GeneAnnotation): The new gene annotation to be resolved.
+            curr_line_num (int): The current line number in the file.
+
+        Returns:
+            GeneAnnotation or None: The resolved gene annotation or None if it cannot be resolved.
```

Comment:
It looks like you also return `None` if the resolution is in favor of the existing gene annotation

Reply:
fixed